odl/util/utility.py

```diff
@@ -163,6 +185,145 @@ def is_complex_floating_dtype(dtype):
     return np.issubsctype(dtype, np.complexfloating)
 
 
+def default_dtype(impl, field):
+    """Return the default data type of different implementations."""
+    from odl.set.sets import RealNumbers
+    if impl == 'numpy':
+        if field == RealNumbers():
+            dtype = np.dtype('float64')
+        else:
+            dtype = np.dtype('complex128')
+    elif impl == 'cuda':
+        if field == RealNumbers():
+            dtype = np.dtype('float32')
+        else:
+            raise NotImplementedError('complex data types not supported in '
+                                      'CUDA.')
+            # dtype = np.dtype('complex64')
+    else:
+        raise ValueError("'impl '{}' not understood.".format(impl))
+
+    return dtype
+
+
+def complex_space(space):
```

Comment:
These are now member functions of `FnBase`, `FunctionSpace` and `DiscreteLp`. I thought about putting it into `Discretization` directly, but there you can have any `LinearSpace` as `uspace`, and it's not possible to turn that into a complex space (unknown init signature).